### PR TITLE
SOT-98: Summary of chapter 12 of the book Proofs, Arguments, and Zero-Knowledge

### PR DESCRIPTION
**Pull request by hduoc2003**
Σ-Protocols and Commitments from Hardness of Discrete Logarithm

# Check List

- [x] Did you check your grammar / reader proof your document using ChatGPT?
- [x] Don't forget to squash commits into meaningful chunks before merging
- [x] After merging this, please update your task status!
- [x] Did you verify that your math formulas are rendered correctly?

### CODE REVIEW

File: terms/abelian.md [hunk 1]
```diff
@@ -0,0 +1 @@
+In cases where the [group](group.md) operation is commutative, the group is called abelian.
```
Comment (sonntuet1997) [inline terms/abelian.md:1]: Missing header

File: terms/commitment_scheme.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+In a commitment scheme, there are two parties, a committer and a verifier. The committer wishes to bind itself to a message without
```
Comment (sonntuet1997) [inline terms/commitment_scheme.md:1]: Missing header

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+$1−1/2^λ$ , where $λ$ denotes the number of bits output by $R$ on any query.
+
+## 12.3 A Homomorphic Commitment Scheme<a id="12.3"></a>
+
+You should read the definition of commitment scheme [here](../../terms/commitment_scheme.md).
+
+**A Perfectly Hiding Commitment Scheme from any $Σ$-Protocol**: Damgård showed how to use any $Σ$-protocol for any hard relation to
+obtain a perfectly hiding, computationally binding commitment scheme. To ensure hiding, Damgård’s transformation does require the
+simulator used to establish HVZK must be able to take as input not only the public input $h$, but also a challenge $e^*$ , and output a
+transcript $(a, e^* ,z)$.
```
Comment (Tranduy1dol) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:107]: math problem

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 4]
```diff
@@ -0,0 +1,285 @@
...
+$$
+\begin{aligned} m^∗ = (m'_1 −m'_2 )·(e−e')^{−1} \quad mod \quad |G|, \\
+z^∗ = (z'_1 −z'_2 )·(e−e')^{−1} \quad mod \quad |G|.
+\end{aligned}
+$$
+
+Then:
+
+$$
+\begin{aligned} g^{m^{*}}\cdot
```
Comment (Tranduy1dol) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:210]: math problem

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+
+$$
+\begin{aligned} c_1' = Com_{g,h}(z_1,z_2) = α · c_1^e , \\
+c_2' = Com_{g,h}(z_3,z_4) = β · c_2^e , \\
+c_3' = Com_{c_1,h}(z_3,z_5) = γ · c_3^e \end{aligned}
+$$
+
+This final equality for $c_3'$ exploits that:
+
+$$ \begin{aligned} c*3^e = g^{em_1m_2} h^{er3} = c_1^{em_2} h^{er_3−er_1m_2} = Com*{c_1,h}(em_2, er_3 −er_1m_2). \end{aligned} $$
```
Comment (Tranduy1dol) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:276]: math problem

File: terms/commitment_scheme.md [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+all keys are public then $ck$ = $vk$), while $Commit$ is a randomized algorithm that takes as input the committing key $ck$ and the
+message $m$ to be committed and outputs the commitment $c$, as well as possibly extra “opening information” $d$ that the committer may
+hold onto and only reveal during the verification procedure. $Verify$ takes as input the commitment, the verification key, and a
+claimed message $m'$ provided by the committer, and any opening information $d$ and decides whether to accept $m$ as a valid opening of
+the commitment.
+
+A commitment scheme is correct if $Verify(vk,Commit(m, ck),m)$ accepts with probability $1$, for any $m$ (i.e., an honest committer can
+always successfully open the commitment to the value that was committed). A commitment scheme is perfectly hiding if the distribution
+of the commitment $Commit(m, ck)$ is independent of $m$. Finally, a commitment scheme is computationally binding if it require
+exorbitant computational power to find some $d', m' \neq m$ such that $Verify(vk,(c,d),m) = Verify(vk,(c,d'),m') = 1$.
```
Comment (sonntuet1997) [inline terms/commitment_scheme.md:18]:
```mermaid
graph LR
A[KeyGen] --> B{ck, vk}
B --> C{Commit}
C --> D{c, d}
C --> E{m}
D --> F{Verify}
F -->|m' = m| G{Accept}
F -->|m' != m| H{Reject}
E --> F
G --> I{Output: success}
H --> I{Output: fail}
```

File: terms/commitment_scheme.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+that generates a commitment key $ck$ and verification key $vk$ that are available to the committer and the verifier respectively (if
+all keys are public then $ck$ = $vk$), while $Commit$ is a randomized algorithm that takes as input the committing key $ck$ and the
+message $m$ to be committed and outputs the commitment $c$, as well as possibly extra “opening information” $d$ that the committer may
+hold onto and only reveal during the verification procedure. $Verify$ takes as input the commitment, the verification key, and a
+claimed message $m'$ provided by the committer, and any opening information $d$ and decides whether to accept $m$ as a valid opening of
+the commitment.
+
+A commitment scheme is correct if $Verify(vk,Commit(m, ck),m)$ accepts with probability $1$, for any $m$ (i.e., an honest committer can
+always successfully open the commitment to the value that was committed). A commitment scheme is perfectly hiding if the distribution
+of the commitment $Commit(m, ck)$ is independent of $m$. Finally, a commitment scheme is computationally binding if it require
```
Comment (sonntuet1997) [inline terms/commitment_scheme.md:17]: highlight: computationally binding, perfectly hiding

File: terms/commitment_scheme.md
Comment (sonntuet1997) [inline terms/commitment_scheme.md:1]: What is the relation between commitment scheme and polynomial commitment?

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 8]
```diff
@@ -0,0 +1,285 @@
+---
+Title: "Σ-Protocols and Commitments from\r Hardness of Discrete Logarithm"
+Status: In Progress
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:3]: In Progress ?

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 9]
```diff
@@ -0,0 +1,285 @@
+---
+Title: "Σ-Protocols and Commitments from\r Hardness of Discrete Logarithm"
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:2]: `\r`?

File: terms/commitment_scheme.md (hunk 6 above)
Comment (sonntuet1997) [inline terms/commitment_scheme.md:18]:
```mermaid
flowchart TD;
    subgraph Setup
        A[KeyGen: Generates ck for committer]
        B[KeyGen: Generates vk for verifier]
    end
    subgraph Committer
        C["Commit: Generates (c, d) using m"]
    end
    subgraph Verifier
        D["Verify: Checks commitment validity with (c, d), vk and m"]
    end
    A -->|ck| C
    B -->|vk| D
    C --> E("(c, d)")
    E -->|d| D
    D -->|Decision| F[Accept or Reject]

```

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 10]
```diff
@@ -0,0 +1,285 @@
...
+  ([Section 12.2.2](#12.2.2)).
+- Allowing the prover to cryptographically commit to group elements without revealing the committed group element to the verifier until
+  later ([Section 12.3](#12.3)).
+- Establishing product relationships between committed values
+  ([Section 12.3.2](#12.3.2))
+
+### 12.2.1 $Σ$-Protocols
+
+A relation $R$ specifies a collection of “valid” instance-witness pairs $(h,w)$. For example, $h=g^w$ for discrete logarithm relation.\
+A $\Sigma$-protocol is a 3-message public coin protocol in which bot verifier and prover know a public input $h$, and prover knows a
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:49]: both

File: terms/commitment_scheme.md (hunk 6 above)
Comment (hduoc2003) [reply terms/commitment_scheme.md:18]:
```mermaid
flowchart TD;
    subgraph Setup
        A[KeyGen: Generates ck for committer]
        B[KeyGen: Generates vk for verifier]
    end
    subgraph Committer
        C["Commit: Generates (c, d) using m"]
    end
    subgraph Verifier
        D["Verify: Checks commitment validity with (c, d), vk and m"]
    end
    A -->|ck| C
    B -->|vk| D
    C --> E("(c, d)")
    E -->|d| D
    D -->|Decision| F[Accept or Reject]
```

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 11]
```diff
@@ -0,0 +1,276 @@
...
+
+## 12.1 Cryptographic Background
+
+### 12.1.1 A Brief Introduction to Groups
+
+This is the definition of [group](../../terms/group.md).
+
+### 12.1.2 The Discrete Logarithm Problem and Background on Elliptic Curves
+
+#### 12.1.2.1 Discrete Log Problem
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:17]: Already have a section for this in discrete_logarithm.md

Comment (sonntuet1997) [reply articles/proofs-arguments-and-zero-knowledge/chapter_12.md:17]: Merge this with it

File: terms/group.md [hunk 12]
```diff
@@ -0,0 +1,18 @@
+# Group
+
+Informally, a group $G$ is any set equipped with an operation that behaves like multiplication. A group must satisfy the following four
```
Comment (sonntuet1997) [inline terms/group.md:3]: is any set equipped with an operation ($\cdot$)

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 13]
```diff
@@ -0,0 +1,276 @@
...
+
+For a specified group $G$ the discrete logarithm problem takes as input two group elements $g$ and $h$, and the goal is to output a
+positive integer $i$ such that $g^i = h$ (if $G$ is of prime order then such an $i$ is guaranteed to exist). The discrete logarithm
+problem is believed to be computationally intractable in certain groups $G$. An important caveat is that quantum computers can solve
+the discrete logarithm problem in polynomial time via Shor’s algorithm. Hence, cryptosystems whose security is based on the assumed
+hardness of the discrete logarithm problem are not post-quantum secure.
+
+#### 12.1.2.2 Elliptic Curve Groups
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:25]: Move this to `elliptic_curve_group.md` and link it with this chapter and the elliptic_curve definition

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 14]
```diff
@@ -0,0 +1,276 @@
...
+
+You should take a look at Elliptic curve [here](../../terms/elliptic_curve.md). \
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:27]: Missing definition of base field

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md (hunk 11 above)
Comment (hduoc2003) [reply articles/proofs-arguments-and-zero-knowledge/chapter_12.md:17]: Should I add the text that implies "discrete logarithm problem are not post-quantum secure" to discrete_logarithm.md?

File: terms/scalar_field.md [hunk 15]
```diff
@@ -0,0 +1,7 @@
+# Scalar field
+
+In the context of [elliptic curve group](elliptic_curve_group.md), **scalar field** $F_r$ is the field of scalars used in the
```
Comment (hduoc2003) [inline terms/scalar_field.md:3]: There is a problem with displaying math formulas here but I don't know how to fix it. I copied the exact same thing to try in another file but it doesn't have the same error.

Comment (sonntuet1997) [reply terms/scalar_field.md:3]: Remove hidden invalid characters.
Screenshot 2024-03-03 at 18 12 08

File: terms/elliptic_curve_group.md [hunk 16]
```diff
@@ -0,0 +1,8 @@
+# Elliptic curve group
```
Comment (sonntuet1997) [inline terms/elliptic_curve_group.md:1]: Re-organize the content of this file. It's not cohesive now.

File: terms/group.md [hunk 17]
```diff
@@ -0,0 +1,18 @@
+# Group
+
+Informally, a group $G$ is any set equipped with an operation $(\cdot)$. A group must satisfy the following four
+properties:
+
+- Closure: the product of two elements in $G$ are also in $G$, i.e., for all $a,b ∈ G$, $a · b$ is also in $G$.
```
Comment (sonntuet1997) [inline terms/group.md:6]: replace `·`  with \cdot

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 18]
```diff
@@ -0,0 +1,271 @@
+---
+Title: "\Sigma-Protocols and Commitments from Hardness of Discrete Logarithm"
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:2]: Screenshot 2024-03-07 at 08 51 43

File: terms/commitment_scheme.md [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+    end
+    A -->|ck| C
+    B -->|vk| D
+    C --> E("(c, d)")
+    E -->|d| D
+    D -->|Decision| F[Accept or Reject]
+```
+
+A commitment scheme is correct if $Verify(vk,Commit(m, ck),m)$ accepts with probability $1$, for any $m$ (i.e., an honest committer can
+always successfully open the commitment to the value that was committed). A commitment scheme is ==perfectly hiding== if the distribution
```
Comment (bjergsen243) [inline terms/commitment_scheme.md:37]: ==perfectly hiding==

Comment (hduoc2003) [reply terms/commitment_scheme.md:37]: > ==perfectly hiding==

It's not a syntax error in Obsidian but in Github, that's the tradeoff

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 20]
```diff
@@ -0,0 +1,271 @@
...
+### 12.1.1 A Brief Introduction to Groups
+
+This is the definition of [group](../../terms/group.md).
+
+### 12.1.2 The Discrete Logarithm Problem and Background on Elliptic Curves
+
+#### 12.1.2.1 Discrete Log Problem
+
+You should read the definition of Discrete Logarithm [here](../../terms/discrete_logarithm.md)
+#### 12.1.2.2 Elliptic Curve Groups
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:20]:
```suggestion

#### 12.1.2.2 Elliptic Curve Groups
```

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 21]
```diff
@@ -0,0 +1,271 @@
...
+
+You should take a look at Elliptic curve group [here](../../terms/elliptic_curve_group.md).
+
+## 12.2 Schnorr’s $\Sigma$-Protocol For Knowledge of Discrete Logarithms
+
+This section covers:
+
+- Establishing that the prover has knowledge of a discrete logarithm of some group element ([Section 12.2.2](#12.2.2)).
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:28]:
```suggestion
- Establishing that the prover has knowledge of a discrete logarithm of some group elements ([Section 12.2.2](#12.2.2)).
```

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 22]
```diff
@@ -0,0 +1,271 @@
...
+- Allowing the prover to cryptographically commit to group elements without revealing the committed group element to the verifier until
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:29]:
```suggestion
- Allowing the prover to cryptographically commit to group elements without revealing them to the verifier until
```

**Review by bjergsen243 — APPROVED** [review]

**Review by Tranduy1dol — APPROVED** [review]

File: terms/group.md [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+- Associativity: for all $a,b, c ∈ G, a  \cdot (b  \cdot  c) = (a  \cdot  b) \cdot  c$.
+- Identity: there an element denoted $1_G ∈ G$ such that $1_G  \cdot  g = g  \cdot  1_G = g$ for all $g ∈ G$
+- Invertibility: For each $g ∈ G$, there is an element $h$ in $G$ such that $g \cdot h = 1_G$. This element $h$ is denoted $g^{−1}$ .
+  An example is the the set of **invertible** matrices, which forms a group under the matrix multiplication operation.
+
+A group $G$ is said to be **cyclic** if there is some group element $g$ such that all group elements can be generated by repeatedly
+multiplying $g$ with itself, i.e., if every element of $G$ can be written as $g^i$ for some positive integer $i$. Such an element of
+$g$ is called a generator for $G$. Any cyclic group is [abelian](abelian.md).
+
+The cardinality $|G|$ is called the order of $G$. For any integer $ℓ$, if $z ≡ ℓ$ mod $|G|,$ then $g^ℓ = g^z$ (and $g^{|G|} = 1_G$).
```
Comment (ktokto313) [inline terms/group.md:16]: what is $z$ in this line

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 24]
```diff
@@ -0,0 +1,271 @@
...
+---
+
+**Protocol 5** Standard presentation of Pedersen commitments in a cyclic group $G$ for which the Discrete Logarithm problem is
+intractable.
+
+---
+
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order. The key generation procedure publishes randomly chosen generators $g,h
+\in G$, which serve as both the commitment key and verification key.  
+$2$: To commit to a number $m \in \{0,...,|G| −1\}$, committer picks a random $z \in \{0,...,|G| −1\}$ and sends $c  \leftarrow{}  g^m
```
Comment (VanhGer) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:120]: \lbrace and \rbrace

File: terms/elliptic_curve.md [hunk 25]
```diff
@@ -32,6 +32,7 @@ The equation defining an elliptic curve is $y^{2} = x^{3} + ax + b$.
 > [!WARNING]  
 > An elliptic curve is not an ellipse.
 
+You should also read about [elliptic curve group](elliptic_curve_group.md).
 ## Terminology
```
Comment (sonntuet1997) [inline terms/elliptic_curve.md:36]: new empty line
```suggestion

## Terminology
```

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 26]
```diff
@@ -0,0 +1,289 @@
...
+
+You should read the definition of Discrete Logarithm [here](../../terms/discrete_logarithm.md)
+
+#### 12.1.2.2 Elliptic Curve Groups
+
+You should take a look at Elliptic curve group [here](../../terms/elliptic_curve_group.md).
+
+## 12.2 Schnorr’s $\Sigma$-Protocol For Knowledge of Discrete Logarithms
+
+This section covers:
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:27]: This should be `the purpose of this sum protocol` or smth like that

Comment (sonntuet1997) [reply articles/proofs-arguments-and-zero-knowledge/chapter_12.md:27]: Screenshot 2024-03-26 at 10 32 17
Screenshot 2024-03-26 at 10 32 27

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 27]
```diff
@@ -0,0 +1,289 @@
...
+and two
+additional properties:
+
+- **Special soundness**: There exists a polynomial time algorithm $Q$ such that, when given as input a pair of accepting transcripts
+  $(a, e, z)$ and $(a, e',z')$ with $e \neq e'$ , $Q$ outputs a witness $w$ such that $(h,w) \in R$.
+- **Honest Verifier Perfect Zero-Knowledge (HVZK)**: There must be a randomized polynomial time simulator that takes as input the
+  public input $h$ from the $\Sigma$-protocol, and outputs a transcript $(a, e,z)$ such that the distribution over transcripts output
+  by the
+  simulator is identical to the distribution over transcripts produced by the honest verifier in the $\Sigma$-protocol interacting with
+  the
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:49]: Add **remark 12.1** section. It's important

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 28]
```diff
@@ -0,0 +1,289 @@
...
+  honest prover.
+
+### 12.2.2 Schnorr’s $\Sigma$-Protocol For the Discrete Logarithm Relation <a id="12.2.2"></a>
+
+The algorithm can be presented as follows:  
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order with generator $g$.
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:55]:
```suggestion
1. Let $G$ be a (multiplicative) cyclic group of prime order with generator $g$.  
```

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 29]
```diff
@@ -0,0 +1,289 @@
...
+  by the
+  simulator is identical to the distribution over transcripts produced by the honest verifier in the $\Sigma$-protocol interacting with
+  the
+  honest prover.
+
+### 12.2.2 Schnorr’s $\Sigma$-Protocol For the Discrete Logarithm Relation <a id="12.2.2"></a>
+
+The algorithm can be presented as follows:  
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order with generator $g$.  
+$2$: Public input is $h$ = $g^w$, where only prover knows $w$.
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:56]: 2.

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 30]
```diff
@@ -0,0 +1,289 @@
...
+  simulator is identical to the distribution over transcripts produced by the honest verifier in the $\Sigma$-protocol interacting with
+  the
+  honest prover.
+
+### 12.2.2 Schnorr’s $\Sigma$-Protocol For the Discrete Logarithm Relation <a id="12.2.2"></a>
+
+The algorithm can be presented as follows:  
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order with generator $g$.  
+$2$: Public input is $h$ = $g^w$, where only prover knows $w$.  
+$3$: Picks a random number $r \in \lbrace 0,...,|G| −1 \rbrace$ and sends $a \leftarrow{} g^r$ to the verifier.
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:57]: 3.

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 31]
```diff
@@ -0,0 +1,289 @@
...
+- **Special soundness**: There exists a polynomial time algorithm $Q$ such that, when given as input a pair of accepting transcripts
+  $(a, e, z)$ and $(a, e',z')$ with $e \neq e'$ , $Q$ outputs a witness $w$ such that $(h,w) \in R$.
+- **Honest Verifier Perfect Zero-Knowledge (HVZK)**: There must be a randomized polynomial time simulator that takes as input the
+  public input $h$ from the $\Sigma$-protocol, and outputs a transcript $(a, e,z)$ such that the distribution over transcripts output
+  by the
+  simulator is identical to the distribution over transcripts produced by the honest verifier in the $\Sigma$-protocol interacting with
+  the
+  honest prover.
+
+### 12.2.2 Schnorr’s $\Sigma$-Protocol For the Discrete Logarithm Relation <a id="12.2.2"></a>
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:52]: <a id="12.2.2"></a> ?

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 32]
```diff
@@ -0,0 +1,289 @@
...
+
+The algorithm can be presented as follows:  
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order with generator $g$.  
+$2$: Public input is $h$ = $g^w$, where only prover knows $w$.  
+$3$: Picks a random number $r \in \lbrace 0,...,|G| −1 \rbrace$ and sends $a \leftarrow{} g^r$ to the verifier.  
+$4$: Verifier responds with a random element $e \in \lbrace 0,...,|G|−1 \rbrace.$  
+$5$: Prover responds with $z \leftarrow{}  (we+r)$ mod $|G|$.  
+$6$: Verifier checks that $a.h^e = g^z$.
+
+Perfect completeness is easy to establish, so we will prove that Schnorr’s protocol satisfies special soundness, and honest-verifier
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:62]:
```suggestion
**Perfect completeness** is easy to establish, so we will prove that Schnorr’s protocol satisfies special soundness, and honest-verifier
```

Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:62]: Write down the formula

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 33]
```diff
@@ -0,0 +1,289 @@
...
+A relation $R$ specifies a collection of “valid” instance-witness pairs $(h,w)$. For example, $h=g^w$ for discrete logarithm relation.\
+A $\Sigma$-protocol is a 3-message public coin protocol in which both verifier and prover know a public input $h$, and prover knows a
+witness $w$ such that $(h,w)\in R$. Let denote the three messages by $(a, e, z)$, with the prover first sending $a$, the verifier
+responding with a challenge $e$ , and the prover replying with $z$. A $\Sigma$-protocol is required to satisfy perfect completeness,
+and two
+additional properties:
+
+- **Special soundness**: There exists a polynomial time algorithm $Q$ such that, when given as input a pair of accepting transcripts
+  $(a, e, z)$ and $(a, e',z')$ with $e \neq e'$ , $Q$ outputs a witness $w$ such that $(h,w) \in R$.
+- **Honest Verifier Perfect Zero-Knowledge (HVZK)**: There must be a randomized polynomial time simulator that takes as input the
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:45]: This is hard to understand

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 34]
```diff
@@ -0,0 +1,289 @@
...
+
+You should read the definition of commitment scheme [here](../../terms/commitment_scheme.md).
+
+**A Perfectly Hiding Commitment Scheme from any $\Sigma$-Protocol**: Damgård showed how to use any $\Sigma$-protocol for any hard
+relation to
+obtain a perfectly hiding, computationally binding commitment scheme. To ensure hiding, Damgård’s transformation does require the
+simulator used to establish HVZK must be able to take as input not only the public input $h$, but also a challenge $e^{ * }$ , and
+output a transcript $(a, e^ * ,z)$.
+
+Here is how Damgård’s commitment scheme works. Let denote $(h,w)  \leftarrow{} Gen$, and declares $h$ to be both the committing key
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:106]: Split this into a subsection:
### Damgård’s commitment scheme

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 35]
```diff
@@ -0,0 +1,289 @@
...
+
+Pedersen commitment scheme obtained from Schnorr’s protocol via Damgård’s transformation:
+
+---
+
+**Protocol 6** Commitment scheme obtained from Schnorr’s protocol via Damgård’s transformation.
+
+---
+
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order.
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:147]: 1.

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 36]
```diff
@@ -0,0 +1,289 @@
...
+of $m$. Below is the traditional presentation of Pedersen commitments:
+
+---
+
+**Protocol 5** Standard presentation of Pedersen commitments in a cyclic group $G$ for which the Discrete Logarithm problem is
+intractable.
+
+---
+
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order. The key generation procedure publishes randomly chosen generators $g,h
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:130]: 1.

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 37]
```diff
@@ -0,0 +1,289 @@
...
+$2$: To commit to a number $m \in \lbrace 0,...,|G| −1 \rbrace$, committer picks a random $z \in \lbrace 0,...,|G| −1 \rbrace$ and
+sends $c \leftarrow{} g^m
+\cdot h^z$ .  
+$3$: To open a commitment $c$, committer sends $(m,z)$. Verifier checks that $c = g^m \cdot h^z$ .
+
+---
+
+Pedersen commitment scheme obtained from Schnorr’s protocol via Damgård’s transformation:
+
+---
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:141]: ?

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 38]
```diff
@@ -0,0 +1,289 @@
...
+
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order. The key generation procedure publishes randomly chosen generators $g,h
+\in G$, which serve as both the commitment key and verification key.  
+$2$: To commit to a number $m \in \lbrace 0,...,|G| −1 \rbrace$, committer picks a random $z \in \lbrace 0,...,|G| −1 \rbrace$ and
+sends $c \leftarrow{} g^m
+\cdot h^z$ .  
+$3$: To open a commitment $c$, committer sends $(m,z)$. Verifier checks that $c = g^m \cdot h^z$ .
+
+---
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:137]: remove these `---`

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 39]
```diff
@@ -0,0 +1,289 @@
...
+
+Pedersen commitment scheme obtained from Schnorr’s protocol via Damgård’s transformation:
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:139]: Should be a header

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 40]
```diff
@@ -0,0 +1,289 @@
...
+
+**Additive Homomorphism**: This means that the verifier can take two commitments $c_1$ and $c_2$, to values $m_1,m_2 \in \lbrace
+0,...,|G| −
+1 \rbrace$ (with $m_1,m_2$ known to the committer but not to the verifier), and the verifier on its own can derive a commitment $c_3$
+to $m_3
+:= m_1 + m_2$, such that the prover is able to open $c_3$ to $m_3$. This is done by simply letting $c_3 \leftarrow{} c_1 \cdot
+c_2$. As for "opening information" provided by the prover, if $c_1 = h^{m_1} \cdot g^{z_1}$ and $c_2 = h^{m_2} \cdot g^{z_2}$ ,
+then $c_3 = h^{m_1+m_2} \cdot g^{z_1+z_2}$, so the opening information for $c_3$ is simply $(m_1 + m_2,z_1 + z_2).$
+
+**Perfect HVZK Proof of Knowledge of Opening**: It will be useful for the prover to prove that it knows how to open a commitment $c$ to
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:167]: New sub section

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 41]
```diff
@@ -0,0 +1,289 @@
...
+some value, without actually opening the commitment. Let's see it:
+
+---
+
+**Protocol 7** Zero-Knowledge Proof of Knowledge of Opening of Pedersen Commitment
+
+---
+
+$1$: Let $G$ be a (multiplicative) cyclic group of prime order over which the Discrete Logarithm relation is hard, with randomly chosen
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:176]: Place algorithm in something like this:

```
abd
```

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 42]
```diff
@@ -0,0 +1,289 @@
...
+$5$: Let $m' \leftarrow{} me + d$ and $z' \leftarrow{} ze + r$, and let $c' \leftarrow{} Com(m ' ,z ' ).$ While Verifier does not
+know $m '$ and $z '$ , Verifier can derive $c '$ unaided from $Com(m,z)$ and $Com(d,r)$ using additive homomorphism: $c'=Com(m,z)^e
+\cdot Com(d,r)$  
+$6$: Prover sends $(m ' ,z' ).$  
+$7$: Verifier checks that $m' ,z'$ is valid opening information for $c'$ , i.e., that $g^{m'} \cdot h^{z'} = c'$ .
+
+---
+
+**Perfect Completeness**: $g^{m'} \cdot h^{z'} = g^{me+d} \cdot h^{ze+r} = c^e \cdot a$
+**Special Soundness**: Given two accepting transcripts $(a, e,(m'_1 ,z'_1 ))$ and $(a, e',(m'_2 ,z'_2 ))$ with $e \neq e'$ , we have to
```
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:206]: separate line

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md
Comment (sonntuet1997) [inline articles/proofs-arguments-and-zero-knowledge/chapter_12.md:1]: General question: What should we remember after reading this chapter?

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md [hunk 43]
```diff
@@ -0,0 +1,289 @@
...
+a\cdot\Big(c^{e^{\prime}}\cdot a\Big)^{-1}\Big)^{(e-e^{\prime})^{-1}}=c \end{aligned} $$
+
+where the penultimate equality follows from the fact that $(a, e,(m'_1 ,z'_1 ))$ and $(a, e',(m'_1 ,z'_1 ))$ are accepting transcripts.
+That is, $(m^* ,z^* )$ is a valid (message, opening information) pair for the commitment $c$.  
+**Perfect HVZK**: The simulator samples $e,m' ,z'$ uniformly at random from $\lbrace 0,...,|G| −1 \rbrace$ and then sets $a
+\leftarrow{} g^{m'}
+\cdot h^{z'} \cdot
+c^{−e}$ and outputs ($a, e,(m' ,z' )).$
+
+### 12.3.2 Establishing A Product Relationship Between Committed Values<a id="12.3.2"></a>
```
[the remark this answers is not in the document]
Comment (hduoc2003) [reply articles/proofs-arguments-and-zero-knowledge/chapter_12.md:227]: This section only writes about one content, so I think it should not be divided into some subsections

File: articles/proofs-arguments-and-zero-knowledge/chapter_12.md (hunk 41 above)
Comment (hduoc2003) [reply articles/proofs-arguments-and-zero-knowledge/chapter_12.md:176]: Place these algorithms in block will raise math formula errors, I don't know a better way to deal with it.

**Review by sonntuet1997 — APPROVED** [review]

**Pull request merged**